pkgs/by-name/im/immich/machine-learning.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  src,
+  immich,
+  python3,
+}:
+python3.pkgs.buildPythonApplication {
+  pname = "immich-machine-learning";
+  inherit (immich) version;
+  src = "${src}/machine-learning";
+  pyproject = true;
+
+  postPatch = ''
+    substituteInPlace pyproject.toml --replace-fail 'fastapi-slim' 'fastapi'
+
+    # Allow immich to use pydantic v2
+    substituteInPlace app/schemas.py --replace-fail 'pydantic' 'pydantic.v1'
+    substituteInPlace app/main.py --replace-fail 'pydantic' 'pydantic.v1'
+    substituteInPlace app/config.py \
+      --replace-fail 'pydantic' 'pydantic.v1'
```

Comment:
These changes were rejected upstream so we should use `packageOverrides` to pin `pydantic_1`.

Reply:
No, that causes basically the whole python world to rebuild. That's just unnecessary waste of hydra resources. For the inference endpoint the patch works and thats the only one used by this setup.

Reply:
> For the inference endpoint the patch works and thats the only one used by this setup.

Why aren't the others used?
And inference apparently doesn't use pydantic, so why isn't the dependency removed outright?

Reply:
Because immich only uses it. The other endpoint is a healthcheck endpoint that is used e.g. in docker

Reply:
What happens when you call the unused endpoints? We don't want shipping the wrong version of pudantic to make a DoS attack possible.

Reply:
> What happens when you call the unused endpoints? We don't want shipping the wrong version of pydantic to make a DoS attack possible.

Reply:
The machine learning service only binds to localhost, if an attacker can DOS that, you have other problems